src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -2290,6 +2294,66 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	setParentContextKeyService(parentContextKeyService: IContextKeyService): void {
 		this._scopedContextKeyService.updateParent(parentContextKeyService);
 	}
+
+	async handleMouseEvent(event: MouseEvent, contextMenu: IMenu): Promise<{ cancelContextMenu: boolean } | void> {
+		// Don't handle mouse event if it was on the scroll bar
+		if (dom.isHTMLElement(event.target) && event.target.classList.contains('scrollbar')) {
+			return { cancelContextMenu: true };
+		}
+
+		// Middle click
+		if (event.which === 2) {
```

Comment:
At some point in time, it would be awesome if we had a constant for `2`... such as "MouseButtons.Middle" :)
That would also make the comment obsolete. I don't know why we adopted this ugly pattern in the codebase...
(same for 1 and 3)

Reply:
`which` is actually deprecated, I think there's a better web API to do it now